### PR TITLE
Add a .config/appearance/current file

### PR DESCRIPTION
**Pull request by scmx**
Containing Dark or Light. For example with Vim on startup it's faster to
read a file than to run the `defaults read` command.

Vimrc before:

```viml
if executable('defaults')
  if system('defaults read -g AppleInterfaceStyle') == "Dark\n"
    set background=dark
  else
    set background=light
  endif
else
  set background=dark
endif
```

Vimrc after:

```viml
if filereadable(expand("~/.config/appearance/current")) && match(readfile(expand("~/.config/appearance/current")), "Light") != -1
  set background=light
else
  set background=dark
endif
```

- [x] Tests
- [x] Refactor
- [x] Create `current` on startup as well, not just when colorscheme changes

### CODE REVIEW

File: Appearance/AppDelegate.swift [hunk 1]
```diff
@@ -61,6 +61,17 @@ class AppDelegate: NSObject, NSApplicationDelegate {
...
+                   theme.colorScheme.description)
+            let filename = FileManager.default.currentColorschemeFile.absoluteURL
+            do {
+                try theme.colorScheme.description.write(to: filename, atomically: true, encoding: String.Encoding.utf8)
+            } catch {
+                os_log("error: %@", error.localizedDescription)
+                self?.alert(title: "Error updating current colorscheme file: \(filename)", body: error.localizedDescription)
+            }
+            os_log("Did update current colorscheme file")
+
```
Comment (alex-ross) [inline Appearance/AppDelegate.swift]: You could att this as a new call to add above with its own closure like this.

```suggestion
        hooksController.add { [weak self] theme in
            os_log("Will update current colorscheme file, colorscheme is %@",
                   theme.colorScheme.description)
            let filename = FileManager.default.currentColorschemeFile.absoluteURL
            do {
                try theme.colorScheme.description.write(to: filename, atomically: true, encoding: String.Encoding.utf8)
            } catch {
                os_log("error: %@", error.localizedDescription)
                self?.alert(title: "Error updating current colorscheme file: \(filename)", body: error.localizedDescription)
            }
            os_log("Did update current colorscheme file")
        }

        hooksController.add { [weak self] theme in
```

Each call to add will add a new hook to the instance of `HooksController`.

**Comment by scmx** [comment]
>Create current on startup as well, not just when colorscheme changes

Any suggestion for how to do that? 🤔 Made an attempt now.

Could maybe add another parameter to hooks controller `add` where you specify when it should run. Or maybe make it so that all hooks always run on start.

**Comment by alex-ross** [comment]
@scmx try to cherry pick this and test https://github.com/alex-ross/Appearance/commit/0bd3ad530163436501787cf5ad1333cdb5b58c97. It may be enough just to add the start after the hooks is set.

**Comment by alex-ross** [comment]
@scmx here is yet another fixup bd39365 with some integration tests as well that prove that it is working now.

**Comment by scmx** [comment]
Works great! 💯

**Comment by scmx** [comment]
I squashed the fixups and set you as author for second commit, by using author and timestamp from the last fixup :-)
(`git rebase -i` + `git commit --amend -C bd39365 --edit`)
End result is identical to #2, but without fixups.

**Comment by alex-ross** [comment]
Thanks @scmx, here is the new release that includes this feature https://github.com/alex-ross/Appearance/releases/tag/1.7.0

**Pull request merged**